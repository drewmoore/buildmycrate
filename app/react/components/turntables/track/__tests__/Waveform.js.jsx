/* globals describe test expect */
import React                   from 'react';
import renderer                from 'react-test-renderer';
import { mount }               from 'enzyme';
import TurntablesTrackWaveform from '../Waveform.js.jsx';
import Fixtures                from '../../../../__tests__/helpers/fixtures.js.es6';
import '../../../../__tests__/helpers/initialize.js.es6';

describe('TurntablesTrackWaveform', () => {
  const props = {
    turntableId: 1,
    playable:    true
  };
  Object.assign(props, Fixtures.tracks.fortySixAndTwo);
  const template = (<TurntablesTrackWaveform {...props} />);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Rendering Props', () => {
    const waveFormDisplay = mount(template);

    test('should display track progress indicator', () => {
      const indicatorId = `waveform-progress-for-${props.turntableId}`;
      expect(waveFormDisplay.find(`#${indicatorId}`)).toHaveLength(1);
    });
  });
});
