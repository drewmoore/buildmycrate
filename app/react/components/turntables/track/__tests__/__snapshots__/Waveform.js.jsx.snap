// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurntablesTrackWaveform DOM Snapshot should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 waveform-container"
    id="waveform-container-for-1"
  >
    <div
      className="relative"
    >
      <div
        className="turntable-waveform-progress"
        id="waveform-progress-for-1"
      />
      <img
        alt="Track Waveform"
        className="turntable-track-waveform-image clickable"
        onClick={[Function]}
        src={undefined}
      />
    </div>
  </div>
</div>
`;
